dds/idl/marshal_generator.cpp

```diff
@@ -1426,14 +1429,13 @@ namespace {
         }
         indent.erase(0, 2);
         be_global->impl_ <<
-          indent << "}\n";
+          indent << "}\n"
+            "  if (discard_flag) {\n"
+            "    strm.set_construction_status(Serializer::ElementConstructionFailure);\n"
+            "    return false;\n"
+            "  }\n"
+            "  return true;\n";
```

Comment:
My suggestion was a mistake in that it didn't go far enough and it's caused some pretty nasty errors in the tests that ran on this PR.

I didn't see/forgot about [this here](https://github.com/objectcomputing/OpenDDS/blob/b03436f9969b5a3430bdbaf2b41d6eaf6c8fad75/dds/idl/marshal_generator.cpp#L1368). It makes the code in the scope loop over each array element, so the change moved what used to be outside the loop  to be inside it, for example:

```C++
  for (CORBA::ULong i0 = 0; i0 < 2; ++i0) {
    // Do deserialization on elements
  } // End of Loop
  if (discard_flag) {
    strm.set_construction_status(Serializer::ElementConstructionFailure);
    return false;
  }
  return true;
```

became

```C++
  for (CORBA::ULong i0 = 0; i0 < 2; ++i0) {
    // Do deserialization on elements
  if (discard_flag) {
    strm.set_construction_status(Serializer::ElementConstructionFailure);
    return false;
  }
  return true;
  } // End of Loop
```

So I think statement needs to be separated again and the existing code from `indent << "}\n"` up to:

```C++
        string indent = "  ";
        NestedForLoops nfl("CORBA::ULong", "i", arr, indent);
```

needs put in it's own scope. So it would look like:

```C++
      } else { // Enum, String, Struct, Array, Sequence, Union
        {
          string indent = "  ";
          NestedForLoops nfl("CORBA::ULong", "i", arr, indent);
          // ...
          be_global->impl_ <<
            indent << "}\n";
        }
        be_global->impl_ <<
          "  if (discard_flag) {\n"
          "    strm.set_construction_status(Serializer::ElementConstructionFailure);\n"
          "    return false;\n"
          "  }\n"
          "  return true;\n";
      }
```

Reply:
I see what you are saying.  I'll take a look into it.